src/ripple/app/tx/impl/Escrow.cpp

```diff
@@ -459,7 +556,39 @@ EscrowFinish::doApply()
         }
     }
 
-    AccountID const account = (*slep)[sfAccount];
+    if (!isXRP(amount))
+    {
+        if (!ctx_.view().rules().enabled(featurePaychanAndEscrowForTokens))
+            return temDISABLED;
+
+        Rate lockedRate = ripple::Rate(slep->getFieldU32(sfTransferRate));
+        auto issuerAccID = amount.getIssuer();
+        auto const xferRate = transferRate(view(), issuerAccID);
+        // update if issuer rate is less than locked rate
+        if (xferRate < lockedRate)
+            lockedRate = xferRate;
```

Comment:
what is the motivation for adding a new transfer rate field in addition to the AccountRoot TransferRate? just curious on the use cases

Reply:
The transfer rate is saved to the line entry and used in subsequent funding/payouts. A full example is [here](https://github.com/XRPLF/rippled/blob/44ea1ac953cde3839173d9ffe6490712f040a1fd/src/test/app/PayChan_test.cpp#L4896).

You can see that alice can create the channel, the issuer changes the rate down, alice can fund at the new rate, however, if the issuer raises the rate, the amount is lock, no more funding opportunities are enabled and the rate stays at the lower rate.

Reply:
When designing this amendment we reasoned that the participants in an Escrow or Paychan that use an IOU balance shouldn't be more disadvantaged than participants in a simple payment for the same currency.

So a way of looking at this is that an Escrow or Paychan is just a really circuitous type of payment. Just because the payment takes a long time or requires lots of steps doesn't mean its parties should be subject to the ever-changing whims and uncertainty of the issuer. Once the funds are locked, then the fee that's payable to the issuer is also locked. However if the issuer decreases the fee then the advantage is to the parties.